src/kmeans.jl

```diff
@@ -1,14 +1,20 @@
 # K-means algorithm
 
-#### Interface
-
-mutable struct KmeansResult{T<:Real} <: ClusteringResult
-    centers::Matrix{T}         # cluster centers (d x k)
+####
```

Comment:
Do you really need to change this style?

Reply:
alternative is 

```julia
struct KmeansResult{T<:AbstractFloat} <: ClusteringResult
    centers::AbstractMatrix{T} # cluster centers (p x k)
    assignments::Vector{Int}   # assignments (n)
    costs::Vector{<:Real}           # cost of the assignments (n)
    counts::Vector{Int}        # number of samples assigned to each cluster (k)
    cweights::Vector{<:Real}       # cluster weights (k)
    totalcost::Real               # total cost (i.e. objective)
    iterations::Int            # number of elapsed iterations
    converged::Bool            # whether the procedure converged
end
```

I think the parametrized type is cleaner (and makes explicit that costs and totalcost have the same type). I don't think the  previous situation makes sense (casting everything to Float64). So I'm in favour of the change. So I guess my preferences are:

1. as in current suggestion
2. as alternative above
3. as before

Reply:
I was talking about the `###`. This introduces changes all over the file, even in places which wouldn't change otherwise.

Reply:
Oh! right sorry ok, reverting to initial style.